python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -379,3 +386,214 @@ def get_logs_for_run(
+        if serialized is None:
+            return RunsFeedCursor(
+                run_cursor=None,
+                backfill_cursor=None,
+                timestamp=None,
+            )
+        parts = serialized.split(_DELIMITER)
+        if len(parts) != 3:
+            raise DagsterInvariantViolationError(f"Invalid cursor for querying runs: {serialized}")
+
+        return RunsFeedCursor(
+            run_cursor=parts[0] if parts[0] else None,
+            backfill_cursor=parts[1] if parts[1] else None,
+            timestamp=float(parts[2]) if parts[2] else None,
+        )
+
+
+def _fetch_runs_not_in_backfill(
+    instance: DagsterInstance,
+    cursor: Optional[str],
+    limit: Optional[int],
+    created_before: Optional[float] = None,
+) -> Sequence[RunRecord]:
+    """Fetches limit RunRecords that are not part of a backfill and were created before a given timestamp."""
+    runs_filter = (
+        RunsFilter(created_before=datetime_from_timestamp(created_before))
+        if created_before
+        else None
+    )
+    if limit is None:
+        new_runs = instance.get_run_records(cursor=cursor, filters=runs_filter)
+        return [run for run in new_runs if run.dagster_run.tags.get(BACKFILL_ID_TAG) is None]
+
+    runs = []
+    while len(runs) < limit:
+        new_runs = instance.get_run_records(limit=limit, cursor=cursor, filters=runs_filter)
+        if len(new_runs) == 0:
+            return runs
+        cursor = new_runs[-1].dagster_run.run_id
+        runs.extend([run for run in new_runs if run.dagster_run.tags.get(BACKFILL_ID_TAG) is None])
+
+    return runs[:limit]
+
+
+def _fetch_backfills_created_before_timestamp(
+    instance: DagsterInstance,
+    cursor: Optional[str],
+    limit: Optional[int],
+    created_before: Optional[float] = None,
+) -> Sequence[PartitionBackfill]:
+    """Fetches limit PartitionBackfills that were created before a given timestamp.
+
+    Note: This is a reasonable filter to add to the get_backfills instance method. However, we should have a
+    more generalized way of adding filters than adding new parameters to get_backfills. So for now, doing this
+    in a separate function.
+    """
+    created_before = created_before if created_before else get_current_timestamp()
+    if limit is None:
+        new_backfills = instance.get_backfills(cursor=cursor, limit=limit)
+        return [
+            backfill for backfill in new_backfills if backfill.backfill_timestamp <= created_before
+        ]
+
+    backfills = []
+    while len(backfills) < limit:
+        new_backfills = instance.get_backfills(cursor=cursor, limit=limit)
+        if len(new_backfills) == 0:
+            return backfills
+        cursor = new_backfills[-1].backfill_id
+        backfills.extend(
+            [
+                backfill
+                for backfill in new_backfills
+                if backfill.backfill_timestamp <= created_before
+            ]
+        )
```

Comment:
were looping here justto support created before filter?

Reply:
yep, it's not ideal. there isn't support for filtering backfills on anything other than status. I want to add more filter support for backfills, but didn't want to expand this PR. Planning to stack a branch that adds more filtering capabilities